proposal/emjames/proposal.rst

```diff
@@ -0,0 +1,237 @@
+            ...
+        def reset(self):
+            ...
+        def render(self):
+            ...
+        def close(self):
+            ...
+
+
+Engineering infrastructure
+==========================
+
+Documentation
+*************
+
+After discovering reStructuredText, documentation will be generated with `Sphinx <https://www.sphinx-doc.org/en/master/usage/installation.html>`_ from reStructuredText documents. Each file will have a matching documentation page describing the file's objective and API in the file.
+
+Code style guides
+*****************
+
+Code guidelines will follow Google's styleguides:
+
+| `Python Style Guide <https://google.github.io/styleguide/pyguide.html>`_
+| `Shell Style Guide <https://google.github.io/styleguide/shellguide.html>`_
+
+At the time of this writting Modelica style guide were not found, therefore one will be written gradually.
+
+
+Version control guidelines
+**************************
+
+The ``Main`` branch will be workable tested code. New sprints will have a new working branch until tests are all passed and ready to merge into ``Main``.
+
+For best practices refer to Chris Beams' Git commit `blog entry <https://chris.beams.io/posts/git-commit/>`_. For brevity, the *properly formated Git commit subject line* bit will be added here.
+
+**A properly formed Git commit subject line should always be able to complete the following sentence:**
+
+
+| If applied, this commit will your subject line here
+| For example:
+|
+| If applied, this commit will ``refactor subsystem X for readability``
+| If applied, this commit will ``update getting started documentation``
+| If applied, this commit will ``remove deprecated methods``
+| If applied, this commit will ``release version 1.0.0``
+| If applied, this commit will ``merge pull request #123 from user/branch``
+|
+| Notice how this doesn’t work for the other non-imperative forms:
+|
+| If applied, this commit will ``fixed bug with Y``
+| If applied, this commit will ``changing behavior of X``
+| If applied, this commit will ``more fixes for broken stuff``
+| If applied, this commit will ``sweet new API methods``
+| Remember: Use of the imperative is important only in the subject line. You can relax this restriction when you’re writing the body.
+
+    -- Chris Beams
+
+
+Testing
+*******
+Testing is to be done with ``pytest``.
+
+Schedule
+========
+
+This schedule is a **very** rough estimate of the effort needed per objective.
+During the semester, each objective will be implemented via `Scrum <https://en.wikipedia.org/wiki/Scrum_(software_development)>`_. The *Objective* column represents the *Product Backlog*, the *Sprint Backlog* will be written in detail during the sprint planning, the *Sprint* will follow (5 days) and on the last day there will be a *Sprint Review*.
+
+.. list-table::
+   :widths: 15 100 50
+   :header-rows: 1
+
+   * - Week No.
+     - Objective
+     - Effort (hours)
+
+   * - 42
+     - Proposal
+     - 12
```

Comment:
It's great to have hour estimates!  After you start the development, please come back and review the numbers, and reschedule your work accordingly.  It's perfectly OK to adjust your goals based on the real progress.